src/ScModel.cpp

```diff
@@ -7,6 +7,12 @@ void ScModel::addStateMachine(const ScName &name, const ScName &initialState)
     if ( !m_states.contains(name) ) {
```

Comment:
Do I understand right, that the problem is that if an initial state is found first, then 1) the state machine is added to `m_states`,  2) this `if` is not entered and 3) it is not linked to the root node? If so, does removing this `if` helps?

Reply:
That could work, I'll have to try it out. I was a bit concerned about if this might mess with the orthogonal regions handling in addState though. Our statemachine does not have them so I can not test that unfortunately.